### PR TITLE
Added WoE:TE Items from EP 14.2

### PR DESCRIPTION
**Pull request by kisuka**
Adds WoE:TE items from episode 14.2

These are official effects / specs.

### CODE REVIEW

File: db/re/item_combo_db.txt [hunk 1]
```diff
@@ -202,6 +202,7 @@
 2777:2778:5479,{ bonus bMaxHP,300; bonus bMatkRate,5; bonus2 bSubEle,Ele_Neutral,5; }
 2779:2780:5482,{ bonus bMatkRate,10; bonus bInt,5; bonus2 bSubRace,RC_Angel,10; }
 2779:2780:5577,{ bonus bMatkRate,10; bonus bInt,5; bonus2 bSubRace,RC_Angel,10; }
+2950:15067:20709:22012,{ bonus2 bExpAddClass,Class_All,10; }
```
Comment (AnnieRuru) [inline db/re/item_combo_db.txt]: where do the `Class_All` come from ?
I can't find them in `doc\item_bonus.txt` or `db\const.txt`

File: db/re/item_db.conf [hunk 2]
```diff
@@ -19578,22 +19575,23 @@ item_db: (
...
 	Refine: false
-	View: 10
+	View: 23
 	Script: <"
 		bonus bInt,25;
 		bonus bDex,25;
-		bonus5 bAutoSpell,AB_SILENTIUM,1,500,BF_SKILL,1;/* Custom IRO */
-		bonus3 bAddEff,Eff_Poison,500,ATF_SKILL;
-		bonus3 bAddEff,Eff_Silence,500,ATF_SHORT;
+		bonus5 bAutoSpell,"AB_SILENTIUM",1,100,BF_MAGIC,1;
```
Comment (AnnieRuru) [inline db/re/item_db.conf]:
```
prontera,155,185,5	script	sdkjfskf	100,{
	dispbottom AB_SILENTIUM +""; // return 2057
}
```
hercules script engine can return skill constants

File: db/re/item_db.conf [hunk 3]
```diff
@@ -21139,11 +21134,9 @@ item_db: (
 	Script: <"
 		bonus bVit,20;
 		bonus bMdef,10;
-		bonus2 bAddRaceTolerance, RC_All, 30;
-		bonus bUnbreakableShield,0;
-		/* When you receive Melee Physical damage, chance of casting Protective Light for 60 seconds. */
-		/* Cancels Stun, Sleep, Curse, Stone Curse, Poison, Blind, Silence, Bleeding, Chaos, and Frozen. */
-		/* Protects the user against such effects for a set duration. */
+		bonus2 bAddRaceTolerance,RC_All,30;
+		bonus3 bAutoSpellWhenHit,"ALL_RAY_OF_PROTECTION",1,50;
```
Comment (AnnieRuru) [inline db/re/item_db.conf]: this line should be commented
https://github.com/HerculesWS/Hercules/blob/master/db/re/skill_db.txt#L1038

File: db/re/item_db.conf [hunk 4]
```diff
@@ -33684,6 +33670,62 @@ item_db: (
...
+	AegisName: "Kvasir_Ring_Blue"
+	Name: "Kvasir Ring Blue"
+	Type: 5
+	Buy: 20000
+	Weight: 100
+	Loc: 136
+	Refine: false
+	Script: <"
+		bonus bMaxSP,30;
+		skill "ECL_SNOWFLIP",1;
```
Comment (AnnieRuru) [inline db/re/item_db.conf]: use constant

File: db/re/item_db.conf [hunk 5]
```diff
@@ -76293,6 +76435,24 @@ item_db: (
 	Script: <" itemheal rand(20,30),0; ">
 },
 {
+	Id: 11557
+	AegisName: "TE_White_Potion"
+	Name: "TE White Potion"
+	Type: 0
+	Buy: 200
+	Weight: 300
+	Script: <" itemheal rand(325,405),0; ">
```
Comment (AnnieRuru) [inline db/re/item_db.conf]: I can't use `Woe_White_Potion` in prontera town, ok
but I can use `TE_White_Potion` in prontera town, is this official ?

File: db/re/item_db.conf [hunk 6]
```diff
@@ -87001,6 +87258,29 @@ item_db: (
...
+		noselltonpc: true
+		nostorage: true
+		nogstorage: true
+		nomail: true
+		noauction: true
+	}
+	Script: <"
+		getrandgroupitem 12888,1;
+		getrandgroupitem 12888,1;
+		getrandgroupitem 12888,1;
```
Comment (AnnieRuru) [inline db/re/item_db.conf]: `getrandgroupitem Siege_Kit_Box, 1;`

**Comment by Jedzkie** [comment]
@AnnieRuru i can quick fix your comments on kisuka commits xD

**Comment by AnnieRuru** [comment]
try give kisuka about 24 hours to respond
since it doesn't affect server performance, and server didn't throw error

File: db/re/item_db.conf (hunk 6 above)
Comment (kisuka) [inline db/re/item_db.conf]: Nothing else in item_db.conf is using consts for getrandgroupitem. I didn't want to clutter up this commit by changing all the getrandgroupitem calls with const to make it uniform. That can be another commit.

File: db/re/item_combo_db.txt (hunk 1 above)
Comment (kisuka) [inline db/re/item_combo_db.txt]: My brain mixed up rathena and hercules script commands.

**Comment by kisuka** [comment]
https://github.com/HerculesWS/Hercules/pull/1002

File: db/re/item_combo_db.txt (hunk 1 above)
Comment (AnnieRuru) [inline db/re/item_combo_db.txt]: I admit I suffer this symptom time to time too XD

Comment (kisuka) [inline db/re/item_combo_db.txt]: @AnnieRuru the worst is when eA habits creep back in. Over a decade of athena development over four different branches plus aegis scripting @.@ my mind has no idea what's going on anymore.

Comment (AnnieRuru) [inline db/re/item_combo_db.txt]: I agree XD

File: db/re/item_db.conf [hunk 7]
```diff
@@ -86918,6 +87078,103 @@ item_db: (
...
+{
+	Id: 12874
+	AegisName: "Frost_Giant_Blood"
+	Name: "Frost Giant Blood"
+	Type: 2
+	Buy: 20
+	Weight: 100
+	Script: <"
+		specialeffect2 EF_POTION_CON;
+		autobonus "{ heal -3000,0; bonus2 bAddRace,RC_Player,100; bonus2 bMagicAddRace,RC_Player,100; }",10,0,0,SI_GVG_GIANT;
```
Comment (AnnieRuru) [inline db/re/item_db.conf]: `Instantly consumes 3,000 of your HP and in return, increases your Physical/Magic damage on player enemies by 100% for 10 seconds.`
it didn't instantly consume my hp

or actually I think we missing `bonus_script` like rathena ?

Comment (kisuka) [inline db/re/item_db.conf]: Yeah, I assumed that's what autobonus was supposed to do? I guess not :<
@MishimaHaruna input?

Comment (AnnieRuru) [inline db/re/item_db.conf]: if this is rathena ->
https://github.com/rathena/rathena/blob/master/doc/script_commands.txt#L5377
`bonus_script "<script code>",<duration>{,<flag>{,<type>{,<status_icon>{,<char_id>}}}};`
```
bonus_script "{ heal -3000,0; bonus2 bAddRace,RC_Player,100; bonus2 bMagicAddRace,RC_Player,100; }, 10, 8, 1, SI_GVG_GIANT;
```
just guessing

autobonus is trigger when you attack something
its not meant for usable items

Comment (dastgir) [inline db/re/item_db.conf]: Wasn't this line supposed to be sc_start?

File: db/re/item_combo_db.txt (hunk 1 above)
Comment (dastgir) [inline db/re/item_combo_db.txt]: We don't have those like rA

File: db/re/item_db.conf (hunk 5 above)
Comment (dastgir) [inline db/re/item_db.conf]: He forgot to add map_zone entries
